uwblueprint/paramedics-react
# Validation for creating/editing duplicate hospitals and ambulances

[Notion ticket](https://www.notion.so/uwblueprintexecs/no-duplicate-resources-8713dd5852f04c0eb67558cf562da051)

Associated backend PR: https://github.com/uwblueprint/paramedics-web/pull/77

## Changes
- Added frontend validation to prevent the creation/editing of hospitals with the same `name` and ambulances with the same `vehicleNumber`
- Removed `GET_AMBULANCE/HOSPITAL_BY_ID` network call since we already cache the results of `GET_ALL_AMBULANCES/HOSPITALS` and a list of ambulances/hospitals is required in the mode `edit` to check if a duplicate exists

## Screenshots
Adding an existing hospital
![image](https://user-images.githubusercontent.com/41495076/99570302-b9e0fa00-299f-11eb-8415-e363ad43b339.png)

Editing the hospital name to an existing hospital
![image](https://user-images.githubusercontent.com/41495076/99570308-bcdbea80-299f-11eb-9071-6bf07c502100.png)

Adding an existing ambulance
![image](https://user-images.githubusercontent.com/41495076/99570314-bf3e4480-299f-11eb-9230-b64eba4263ba.png)

Editing the ambulance vehicleNumber to an existing ambulance
![image](https://user-images.githubusercontent.com/41495076/99570317-c1080800-299f-11eb-861c-47ed5fd04929.png)

## Testing
- Create a hospital that already exists - result: modal should pop up
- Edit an existing hospital to another hospital that already exists - result: modal should pop up
- Create an ambulance that already exists - result: modal should pop up
- Edit an existing ambulance to another ambulance that already exists - result: modal should pop up
- Creating/editing hospitals and ambulances that do not already exist should work as expected

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
